src/librustc/hir/map/mod.rs

```diff
@@ -130,8 +130,11 @@ fn is_body_owner<'hir>(node: Node<'hir>, hir_id: HirId) -> bool {
 }
 
 pub(super) struct HirOwnerData<'hir> {
-    pub(super) signature: Option<&'hir HirOwner<'hir>>,
-    pub(super) with_bodies: Option<&'hir mut HirOwnerItems<'hir>>,
+    pub(super) signature: Option<&'hir Owner<'hir>>,
+
+    // FIXME(eddyb) make collection per-owner so that `&mut OwnerNodes`
+    // only exists on the stack, and this field is `&OwnerNodes`.
```

Comment:
Not sure what you mean by this FIXME. Using `&mut` here isn't a problem.

Reply:
Fair, I'll remove it.